src/pybind/mgr/volumes/fs/async_job.py

```diff
@@ -165,10 +197,33 @@ def run(self):
                     # Increase concurrency: create more threads.
                     log.debug("creating new threads to job increase")
                     for i in range(c, self.nr_concurrent_jobs):
-                        self.threads.append(JobThread(self, self.vc, name="{0}.{1}.{2}".format(self.name_pfx, time.time(), i)))
-                        self.threads[-1].start()
+                        if not self.run_event.is_set():
+                            break
+                        self.spawn_new_thread(i)
                 self.cv.wait(timeout=5)
 
+    def pause(self):
+        self.run_event.clear()
+
+        self.saved_q = deepcopy(self.q)
```

Comment:
I don't get it? Why do this need to make a copy here?

Reply:
In `volume.py`, we call `AsyncJobs.queue_job()` to queue the clone or purge job. And when we cancel all the jobs without finishing them in `AsyncJobs.pause()`, all volumes/jobs are removed from `self.q`/`self.jobs`. Thus, `self.q` needs to to re-instated when we resume the worker threads. That's what this code does.

Reply:
Marking this comment resolved since it was discussed and changes have moved forward since.